src/motor-interface-pru.p

```diff
@@ -202,14 +216,30 @@ QUEUE_READ:
 	ZERO &mstate, SIZE(mstate)	; clear the motor states
 	ZERO &r3, 4			; initialize delay calculation state register.
 
+	;; STATUS REGISTER
+	;; ! We are assuming that writing the 4 bytes status register is atomic
+	;; We sum the 3 loop counters inside the last 24 bits of a 4 byte variable
+	;; at the end of the ring buffer. At each step loop executed this counter
+	;; is decreased of one unit
+	ADD r28, r28, travel_params.loops_accel
+	ADD r28, r28, travel_params.loops_travel
+	ADD r28, r28, travel_params.loops_decel
+	;; Add one more because the first delay is pre-calculated
```

Comment:
mmh, can you elaborate ? Is it always one ?
This is not immediately clear to me why that is the case.